ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -279,33 +292,70 @@ private[commands] class CommandTracker[Context](
       }
 
       private def getOutputForCompletion(completion: Completion) = {
-        val (commandId, errorText) = {
+        val (potentialCommandKey, errorText) = {
           completion.status match {
             case Some(StatusProto(code, _, _, _)) if code == Status.Code.OK.value =>
-              completion.commandId -> "successful completion of command"
+              TrackedCommandKey(
+                completion.submissionId,
+                completion.commandId,
+              ) -> "successful completion of command"
             case _ =>
-              completion.commandId -> "failed completion of command"
+              TrackedCommandKey(
+                completion.submissionId,
+                completion.commandId,
+              ) -> "failed completion of command"
+          }
+        }
+
+        val commandKey = if (potentialCommandKey.submissionId.isEmpty) {
```

Comment:
Why is it OK if we get completions with empty submission IDs? Also, I'd prefer avoiding constructing invalid keys and throwing a little earlier.

Reply:
I assume this is for the mutable schema. 
Considering we have this check I would also add some validation in the `registerSubmission` where we validate that the submission id is not empty, so that we don't get up to this point if somehow the submission id was empty from the command.

Reply:
The submission id there is not empty, as it is freshly generated and passed in the request. I think that's the only place we can find out whether it's the mutable schema.